nf_core/download.py

```diff
@@ -167,95 +166,187 @@ def download_workflow(self):
+            for r in self.wf_releases:
+                if r["tag_name"] == self.release:
+                    self.wf_sha = r["tag_sha"]
+                    break
+
+            # Can't find the release or branch - throw an error
+            else:
+                log.info(
+                    "Available {} releases: '{}'".format(
+                        self.pipeline, "', '".join([r["tag_name"] for r in self.wf_releases])
+                    )
+                )
+                log.info("Available {} branches: '{}'".format(self.pipeline, "', '".join(self.wf_branches.keys())))
+                raise AssertionError(
+                    "Not able to find release / branch '{}' for {}".format(self.release, self.pipeline)
+                )
+
+        # Set the outdir
+        if not self.outdir:
+            self.outdir = "{}-{}".format(self.pipeline.replace("/", "-").lower(), self.release)
+
+        # Set the download URL and return
+        self.wf_download_url = "https://github.com/{}/archive/{}.zip".format(self.pipeline, self.wf_sha)
+
+    def prompt_container_download(self):
+        """Prompt whether to download container images or not"""
+
+        if self.container is None:
+            stderr.print("\nIn addition to the pipeline code, this tool can download software containers.")
+            self.container = questionary.select(
+                "Download software container images:",
+                choices=["none", "singularity"],
+                style=nf_core.utils.nfcore_question_style,
+            ).unsafe_ask()
+
+    def prompt_use_singularity_cachedir(self):
+        """Prompt about using $NXF_SINGULARITY_CACHEDIR if not already set"""
+        if (
+            self.container == "singularity"
+            and os.environ.get("NXF_SINGULARITY_CACHEDIR") is None
+            and stderr.is_interactive  # Use rich auto-detection of interactive shells
+        ):
+            stderr.print(
+                "\nNextflow and nf-core can use an environment variable called [blue]$NXF_SINGULARITY_CACHEDIR[/] that is a path to a directory where remote Singularity images are stored. "
+                "This allows downloaded images to be cached in a central location."
+            )
+            if rich.prompt.Confirm.ask(
+                f"[blue bold]?[/] [bold]Define [blue not bold]$NXF_SINGULARITY_CACHEDIR[/] for a shared Singularity image download folder?[/]"
+            ):
+                # Prompt user for a cache directory path
+                cachedir_path = None
+                while cachedir_path is None:
+                    cachedir_path = os.path.abspath(
+                        questionary.path(
+                            "Specify the path:", only_directories=True, style=nf_core.utils.nfcore_question_style
+                        ).unsafe_ask()
+                    )
+                    if cachedir_path == os.path.abspath(""):
+                        log.error(f"Not using [blue]$NXF_SINGULARITY_CACHEDIR[/]")
```

Comment:
So if you define your current working directory it doesn't work, right?
In that case I would suggest to add this to the error message.

Reply:
I haven't tested, but assume it returns what is shown? Probably best to delay the os.path stuff and check and return if answer is empty (user wants to get away), then do os.path.abspath(os.path.expanduser(...)) and continue from there (and I'd also suggest asking if we should create if the path does not exist).

Reply:
Yup so `os.path.abspath("")` returns the current working directory. If you define that as your cache dir, the error is thrown. That might be a desired behaviour, but we should adapt the error message to make it clear why the error is thrown I think :) 

![image](https://user-images.githubusercontent.com/15126788/117447159-6bb7b700-af3d-11eb-9087-b48289c9aea2.png)

Reply:
I'd say having a notice is definitely the wrong solution :)

We could have a default provided (to the current directory, e.g. `os.abspath("")`?) or clearly state the empty answer means aborting.

Either way, comparing with anything processed through `os.abspath` seems wrong to me.

Reply:
I suspect that what I meant to do was `if cachedir_path == "":` and included the `abspath` stuff by accident..

Reply:
No, wait, I see why this is needed. The questionary prompt calls `os.path.abspath()` on the returned value, so if the user hits enter with no entry we don't get `""`, we get `os.path.abspath("")`. My intention is the same though - to cancel and end the `while` loop if the user doesn't enter any input.

I don't think that I realised that `os.path.abspath("")` returns the current working directory and presumably never tried passing the cwd as a string. So this is unintended behaviour. I will refactor so that we check for an empty string _before_ calling `os.path.abspath` to avoid this.

Reply:
I was wondering why we haven't hit the `expanduser` thing before, but I guess it's because when stuff is supplied on the command line, bash expands the path for us before it gets to the code. So it's only needed when prompting from within Python and getting the raw path string..

Reply:
Updated 👍🏻